### DISCUSSION TITLE
Calling reusable actions created over GHES from GHEC

### DISCUSSION THREAD

**Discussion by amartyaroy** [community/community#73230] [Enterprise]
### Select Topic Area

General

### Body

Hi 
I have an usecase regarding github actions dont know whether its possible also or not.
We have two instances of github one is GHES which is managed and privately handled by our organisation and the other is over github.com which GHEC. We have some actions which are created over the enterprise org and now we want to call these actions from our github.com orgs.I have tried to call the workflows but it seems its not able to recognise where to find the workflow and returning error of file not found etc.. Can anyone please suggest if anyone came across such scenario or not. If yes, then how to handle it.

**Comment by onurtufekci34** [comment]
The scenario you're describing involves using GitHub Actions across two different GitHub instances: GitHub Enterprise Server (GHES) and GitHub.com (GHEC). It's indeed possible to use actions defined in your GHES organization within repositories in your GitHub.com organization, but there are some considerations and configurations to keep in mind.

Here's how you can set this up:

1. **Action Repositories**:
   - **GitHub.com (GHEC)**: In your GitHub.com organization, create a public or private repository to host the workflow that will call the actions.
   - **GitHub Enterprise Server (GHES)**: In your GHES organization, create repositories that contain the custom actions you want to use.

2. **Access Permissions**:
   - Ensure that the GitHub.com organization has the necessary access permissions to read the repositories on your GitHub Enterprise Server (GHES). You may need to set up appropriate user or team permissions.

3. **Use `uses` Directive**:
   - In your workflow file in the GitHub.com repository (GHEC), use the `uses` directive to specify the actions from the GHES repository. The format should be like this:

   ```yaml
   jobs:
     my_job:
       runs-on: ubuntu-latest
       steps:
         - name: Checkout code
           uses: actions/checkout@v2
           
         - name: Use custom action from GHES
           uses: <GHES_ORG>/<GHES_REPO>/.github/actions/my-action
   ```

   Replace `<GHES_ORG>`, `<GHES_REPO>`, and `my-action` with the appropriate values for your GHES organization and repository.

4. **Workflow File Location**:
   - Ensure that the workflow file in the GitHub.com repository is located in a path where it can access the actions in the GHES repository. The structure should match the specified `uses` directive.

5. **Workflow Trigger**:
   - You can trigger the workflow in your GitHub.com repository as needed, and it should be able to use the actions from the GHES repository as long as the permissions and paths are correctly set up.

6. **GitHub Token and Secrets**:
   - Make sure to handle any necessary secrets or authentication tokens properly, especially if the actions in the GHES repository require access to specific resources or services.

By following these steps, you can set up GitHub Actions workflows in your GitHub.com organization that utilize custom actions defined in your GHES organization. This allows you to share and reuse actions between different GitHub instances while maintaining access control and security.
